twitter/Models/Tweet.m

```diff
@@ -0,0 +1,64 @@
+//
+//  Tweet.m
+//  twitter
+//
+//  Created by Iris Fu on 6/20/22.
+//  Copyright © 2022 Emerson Malca. All rights reserved.
+//
+
+#import "Tweet.h"
+#import "User.h"
+@implementation Tweet
+
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+    self = [super init];
+
+    if (self) {
+        // Is this a re-tweet?
+        NSDictionary *originalTweet = dictionary[@"retweeted_status"];
+        if (originalTweet != nil) {
+            NSDictionary *userDictionary = dictionary[@"user"];
+            self.retweetedByUser = [[User alloc] initWithDictionary:userDictionary];
+
+            // Change tweet to original tweet
+            dictionary = originalTweet;
+        }
+        self.idStr = dictionary[@"id_str"];
+        self.text = dictionary[@"full_text"];
+        self.favoriteCount = [dictionary[@"favorite_count"] intValue];
+        self.favorited = [dictionary[@"favorited"] boolValue];
+        self.retweetCount = [dictionary[@"retweet_count"] intValue];
+        self.retweeted = [dictionary[@"retweeted"] boolValue];
+
+        // initialize user
+        NSDictionary *user = dictionary[@"user"];
+        self.user = [[User alloc] initWithDictionary:user];
+
+        // Format and set createdAtString
+        // Format createdAt date string
+        NSString *createdAtOriginalString = dictionary[@"created_at"];
+        NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+        // Configure the input format to parse the date string
+        formatter.dateFormat = @"E MMM d HH:mm:ss Z y";
+        // Convert String to Date
+        NSDate *date = [formatter dateFromString:createdAtOriginalString];
+        // Configure output format
+        formatter.dateStyle = NSDateFormatterShortStyle;
+        formatter.timeStyle = NSDateFormatterNoStyle;
+        // Convert Date to String
+        self.createdAtString = [formatter stringFromDate:date];
+    }
+    return self;
+}
+
++ (NSMutableArray *)tweetsWithArray:(NSArray *)dictionaries {
```

Comment:
Still on the topic of type safety, a good idea is to use generics whenever possible. Even though Objective-C's implementation of generics is rudimentary, it can still help improving type safety and preventing crashes. Also, avoid the use of mutable types in method signatures unless really necessary.

```
+ (NSArray<Tweet *> * _Nonnull)tweetsWithArray:(NSArray<NSDictionary *> * _Nullable)dictionaries {
```